include/tins/sniffer.h

```diff
@@ -45,6 +45,9 @@
 
 #include <pcap.h>
 
+using pcap_sniffing_method_t = int(*)(pcap_t*, int, pcap_handler, u_char*);
```

Comment:
This should probably be inside `BaseSniffer` and something like `PcapSniffingMethod` should work. Also, I would actually do a hybrid of your initial version and this (using an enum). The reason is that we could have more methods (don't know if it's worth it) that way, also not being limiting with the interface that wants to be provided. So that way you don't even need to know which specific pcap function you want to grab the pointer from, but just look at the enum and use that.